main/parse.h

```diff
@@ -125,7 +125,9 @@ struct sParserDefinition {
 	stringList* currentPatterns;   /* current list of file name patterns */
 	stringList* currentExtensions; /* current list of extensions */
 	stringList* currentAliases;    /* current list of aliases */
-	unsigned int anonumousIdentiferId; /* managed by anon* functions */
+	unsigned int anonymousIdentiferId; /* managed by anon* functions */
+	parserDefinition *parsersAnonReset; /* list of parsers which anonymousIdentiferId
+					       is set for the current input. */
```

Comment:
BTW, does it make sense to implement it like that?  It seems a bit overcomplicated to me, couldn't we just have an array indexed on lang?